packages/opentelemetry-sdk-trace-base/src/export/SimpleSpanProcessor.ts

```diff
@@ -57,21 +57,31 @@ export class SimpleSpanProcessor implements SpanProcessor {
       return;
     }
 
-    internal
-      ._export(this._exporter, [span])
-      .then((result: ExportResult) => {
-        if (result.code !== ExportResultCode.SUCCESS) {
-          globalErrorHandler(
-            result.error ??
-              new Error(
-                `SimpleSpanProcessor: span export failed (status ${result})`
-              )
-          );
-        }
-      })
-      .catch(error => {
-        globalErrorHandler(error);
-      });
+    const doExport = () =>
+      internal
+        ._export(this._exporter, [span])
+        .then((result: ExportResult) => {
+          if (result.code !== ExportResultCode.SUCCESS) {
+            globalErrorHandler(
+              result.error ??
+                new Error(
+                  `SimpleSpanProcessor: span export failed (status ${result})`
+                )
+            );
+          }
+        })
+        .catch(error => {
+          globalErrorHandler(error);
+        });
+
+    // Avoid scheduling a promise to make the behavior more predictable and easier to test
+    if (span.resource.asyncAttributesHaveResolved()) {
+      void doExport();
+    } else {
```

Comment:
The `forceFlush` function is implemented as
```js
  forceFlush(): Promise<void> {
    // do nothing as all spans are being exported without waiting
    return Promise.resolve();
  }
```

If the `span.resource.waitForAsyncAttributes()` is awaited, then the assumption in `forceFlush` is no longer valid, and calling `forceFlush` might return without that span being exported.